core/tests/ledger_cleanup.rs

```diff
@@ -0,0 +1,296 @@
+        let assert_compaction = read_env("ASSERT_COMPACTION", false);
+
+        BenchmarkConfig {
+            benchmark_slots,
+            batch_size,
+            max_ledger_slots,
+            cleanup_blocktree,
+            emit_cpu_info,
+            assert_compaction,
+        }
+    }
+
+    fn emit_header() {
+        println!("EPOCH_MS,DELTA_MS,START_SLOT,BATCH_SIZE,ENTRIES,MAX,SIZE,DELTA_SIZE,CPU_USER,CPU_SYSTEM,CPU_IDLE");
+    }
+
+    fn emit_stats(
+        time_previous: &mut SystemTime,
+        storage_previous: &mut u64,
+        start_slot: u64,
+        batch_size: u64,
+        entries: u64,
+        max_slots: i64,
+        blocktree: &Blocktree,
+        cpu: &CpuStatsInner,
+    ) {
+        let time_now = SystemTime::now();
+        let storage_now = blocktree.storage_size().unwrap_or(0);
+        let (cpu_user, cpu_system, cpu_idle) = (cpu.cpu_user, cpu.cpu_system, cpu.cpu_idle);
+
+        println!(
+            "{},{},{},{},{},{},{},{},{},{},{}",
+            time_now
+                .duration_since(SystemTime::UNIX_EPOCH)
+                .unwrap()
+                .as_millis(),
+            time_now.duration_since(*time_previous).unwrap().as_millis(),
+            start_slot,
+            batch_size,
+            entries,
+            max_slots,
+            storage_now,
+            storage_now as i64 - *storage_previous as i64,
+            cpu_user,
+            cpu_system,
+            cpu_idle,
+        );
+
+        *time_previous = time_now;
+        *storage_previous = storage_now;
+    }
+
+    #[test]
+    fn test_ledger_cleanup_compaction() {
+        let blocktree_path = get_tmp_ledger_path!();
+        let blocktree = Arc::new(Blocktree::open(&blocktree_path).unwrap());
+        let config = get_benchmark_config();
+        eprintln!("BENCHMARK CONFIG: {:?}", config);
+        eprintln!("LEDGER_PATH: {:?}", &blocktree_path);
+
+        let benchmark_slots = config.benchmark_slots;
+        let batch_size = config.batch_size;
+        let max_ledger_slots = config.max_ledger_slots;
+        let batches = benchmark_slots / batch_size;
+
+        let (sender, receiver) = channel();
+        let exit = Arc::new(AtomicBool::new(false));
+        let cleaner =
+            LedgerCleanupService::new(receiver, blocktree.clone(), max_ledger_slots, &exit);
+
+        let exit_cpu = Arc::new(AtomicBool::new(false));
+        let sys = CpuStatsUpdater::new(&exit_cpu);
+
+        let mut time_previous = SystemTime::now();
+        let mut storage_previous = 0;
+
+        emit_header();
+        emit_stats(
+            &mut time_previous,
+            &mut storage_previous,
+            0,
+            0,
+            0,
+            0,
+            &blocktree,
+            &sys.get_stats(),
+        );
+
+        for i in 0..batches {
+            let x = i * batch_size;
+
+            let (shreds, _) = make_many_slot_entries(x, batch_size, ENTRIES_PER_SLOT);
```

Comment:
This just makes empty ticks, I think we should fill these entries up with dummy tx data to add a lot more disk usage. Also entries_per_slot should definitely be more than 1. Right now this tests the best case scenario for blocktree(many empty slots), we wanna hit it with the worst case.

Reply:
(changed `ENTRIES_PER_SLOT` to 50_000, made `ENTRIES_PER_SLOT` configurable)